actionmetadata_test.go

```diff
@@ -122,3 +134,31 @@ func TestKnowledgeBase(t *testing.T) {
 		t.Fail()
 	}
 }
+
+func doesActionRepoExist(filePath string) *github.Response {
+	var tagOrBranch string = "master"
+	splitOnSlash := strings.Split(filePath, "/")
+	owner := splitOnSlash[1]
+	repo := splitOnSlash[2]
+
+	PAT := os.Getenv("PAT")
+
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: PAT},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+
+	client := github.NewClient(tc)
+
+	_, res, err := client.Git.GetRef(context.Background(), owner, repo, fmt.Sprintf("tags/%s", tagOrBranch))
```

Comment:
By using `go-github client` rate limit is exceeding

![image](https://user-images.githubusercontent.com/73299843/151124810-892ff92a-4fd2-4e75-b613-5256b378c9b3.png)